recipes/natural_language_processing/chatbot/Makefile

```diff
@@ -1,55 +1,94 @@
+	ifeq ($(ARCH),amd64)
+		CHROMEDRIVER_DOWNLOAD_PATH := chromedriver_mac64.zip
+	else ifeq ($(ARCH),arm64)
+		CHROMEDRIVER_DOWNLOAD_PATH := chromedriver_mac64_m1.zip
+	endif
+else ifeq ($(OS),Linux)
+	CHROMEDRIVER_DOWNLOAD_PATH := chromedriver_linux64.zip
+endif
+
+.PHONY: install
+install:
+	@if [[ -z "$(CHROMEDRIVER_EXISTS)" ]]; then \
+		if [[ -n "$(UNZIP_EXISTS)" ]]; then \
+			curl -sLO $(CHROMEDRIVER_MIRROR)/$(CHROMEDRIVER_VERSION)/$(CHROMEDRIVER_DOWNLOAD_PATH); \
+			unzip $(CHROMEDRIVER_DOWNLOAD_PATH); \
+			mv chromedriver /usr/local/bin/chromedriver; \
+		elif [[ -z "$(UNZIP_EXISTS)" ]]; then \
+			echo "Install make target requires unizp binary."; \
+		fi; \
+	fi; \
+	pip install -r tests/requirements.txt
+
 .PHONY: build
 build:
-	podman build -f builds/Containerfile -t ${APPIMAGE} .
+	podman build -f builds/Containerfile -t ${APP_IMAGE} .
 
 .PHONY: bootc
 bootc: quadlet
-	podman build $${FROM:+--from $${FROM}} --cap-add SYS_ADMIN --build-arg "SSHPUBKEY=$(SSHPUBKEY)" -f bootc/Containerfile -t ${BOOTCIMAGE} .
+	podman build $${FROM:+--from $${FROM}} --cap-add SYS_ADMIN --build-arg "SSH_PUBKEY=$(SSH_PUBKEY)" -f bootc/Containerfile -t ${BOOTC_IMAGE} .
 
 .PHONY: quadlet
 quadlet:
 	# Modify quadlet files to match the server, model and app image
 	mkdir -p build
-	sed -e "s|SERVERIMAGE|${SERVERIMAGE}|" \
-	    -e "s|APPIMAGE|${APPIMAGE}|g" \
-	    -e "s|MODELIMAGE|${MODELIMAGE}|g" \
+	sed -e "s|SERVER_IMAGE|${SERVER_IMAGE}|" \
+	    -e "s|APP_IMAGE|${APP_IMAGE}|g" \
+	    -e "s|MODEL_IMAGE|${MODEL_IMAGE}|g" \
 	    quadlet/${APP}.image \
 	    > build/${APP}.image
-	sed -e "s|SERVERIMAGE|${SERVERIMAGE}|" \
-	    -e "s|APPIMAGE|${APPIMAGE}|g" \
-	    -e "s|MODELIMAGE|${MODELIMAGE}|g" \
+	sed -e "s|SERVER_IMAGE|${SERVER_IMAGE}|" \
+	    -e "s|APP_IMAGE|${APP_IMAGE}|g" \
+	    -e "s|MODEL_IMAGE|${MODEL_IMAGE}|g" \
 	    quadlet/${APP}.yaml \
 	    > build/${APP}.yaml
 	cp quadlet/${APP}.kube build/${APP}.kube
 
-.PHONY: install
-install:
-	wget https://www.slimjetbrowser.com/chrome/files/103.0.5060.53/google-chrome-stable_current_amd64.deb
-	sudo dpkg -i google-chrome-stable_current_amd64.deb
-	wget https://chromedriver.storage.googleapis.com/103.0.5060.53/chromedriver_linux64.zip
-	unzip chromedriver_linux64.zip
-	pip install -r tests/requirements.txt
-
 .PHONY: run
 run: 
-	podman run -it -p 8501:8501 -e MODEL_SERVICE_ENDPOINT=http://10.88.0.1:8001/v1 ghcr.io/ai-lab-recipes/${APP}
+	podman run --rm -d -it -p $(PORT):$(PORT) -e MODEL_SERVICE_ENDPOINT=http://10.88.0.1:8001/v1 $(APP_IMAGE)
 
 .PHONY: functional_tests
 functional_tests:
-	python3 -m pytest -vvv --driver=Chrome --driver-path=./chromedriver tests/functional
+	pytest -vvv --driver=Chrome --driver-path=$(CHROMEDRIVER_EXISTS) tests/functional
 
 .PHONY: integration_test
 integration_tests:
-	URL=${URL} python3 -m pytest -vvv --driver=Chrome --driver-path=./chromedriver tests/integration
+	URL=${URL} pytest -vvv --driver=Chrome --driver-path=$(CHROMEDRIVER_EXISTS) tests/integration
```

Comment:
This will blow up it CHROMEDRIVER does not exists?

Reply:
Yes it does, good catch, i'll fix this.